### PR TITLE
Raise `ValueError` when concentration values are out of bounds

### PR DESCRIPTION
**Pull request by nollety**
# Description

Resolves eradiate/eradiate-issues#167

# Checklist

- [x] The code follows the relevant coding guidelines
- [x] The code generates no new warnings
- [x] The code is appropriately documented
- [x] The code is tested to prove its function
- [x] The feature branch is rebased on the current state of the `main` branch
- [x] I updated the change log if relevant
- [x] I give permission that the Eradiate project may redistribute my contributions under the terms of its license

### CODE REVIEW

**Comment by leroyvn** [comment]
Hi @nollety, the fix you're suggesting effectively doubles the number of calls to `interp()` for this particular step and I'm concerned with a potential impact on performance. Would you mind checking how this impacts the runtime of the test suite? (See *e.g.* #229 for a quick & dirty time benchmarking method)

**Comment by nollety** [comment]
### Benchmarck results:

**main**
test_norescale: 3.51 s ± 0.0892 (100 reps)
test_rescale: 5.34 s ± 0.0981 (100 reps)

**167-fix**
test_norescale: 3.51 s ± 0.0617 (100 reps)
test_rescale: 5.21 s ± 0.149 (100 reps)

<details>
<summary>Click to expand</summary>

import timeit
import numpy as np

import eradiate

eradiate.set_mode("mono")

from eradiate import unit_registry as ureg
from eradiate.scenes.atmosphere import MolecularAtmosphere


def benchmark(func, n_runs=1000, n_reps=10):
    durations = timeit.Timer(func).repeat(repeat=n_reps, number=n_runs)
    results = {
        "durations": durations,
        "mean": np.mean(durations),
        "std": np.std(durations),
    }
    return results


def test_norescale(n_samples=10000):
    atmosphere = MolecularAtmosphere.afgl_1986()


def test_rescale(n_samples=10000):
    atmosphere = MolecularAtmosphere.afgl_1986(
        concentrations={"H2O": 20.0 * ureg.kg / ureg.m**2}
    )


n_runs = 1000
n_reps = 100


for title, task in (("test_norescale", test_norescale), ("test_rescale", test_rescale)):
    results = benchmark(
        lambda: task(),
        n_runs=n_runs,
        n_reps=n_reps,
    )
    print(
        f"{title:10s}: {results['mean']:.3g} s ± {results['std']:.3g} ({n_reps} reps)"
    )


</details>

File: CHANGELOG.md [hunk 1]
```diff
@@ -54,6 +54,8 @@ when necessary—we also advise to not ignore `DeprecationWarning`s.
   spectrum to cover the full wavelength range of Eradiate ({ghpr}`233`).
 * Fixed a bug where converting an integer to a `Spectrum` would fail 
   (`ghpr`{236}). 
+* Raise an exception when molecular concentrations are out of bounds
+  (`ghpr`{237})
```
Comment (leroyvn) [inline CHANGELOG.md:60]:
```suggestion
  (`ghpr`{237}).
```

File: src/eradiate/radprops/_afgl1986.py [hunk 2]
```diff
@@ -128,7 +128,14 @@ def eval_sigma_a_ckd(self, *bindexes: Bindex, bin_set_id: str) -> pint.Quantity:
...
                     ds.k.sel(bd=(bindex.bin.id, bindex.index))
-                    .interp(z=z, **concentrations, kwargs=dict(fill_value=0.0))
+                    .interp(
+                        z=z,
+                        kwargs=dict(fill_value=0.0),
+                    )  # extrapolate to 0.0 for altitude out of bounds
+                    .interp(
+                        **concentrations,
+                        kwargs=dict(bounds_error=True),
+                    )  # raise when concentration are out of bounds
```
Comment (leroyvn) [inline src/eradiate/radprops/_afgl1986.py:138]: More explicit IMO:
```suggestion
                        kwargs=dict(bounds_error=True),  # raise when concentration are out of bounds
                    )
```

**Review by leroyvn — APPROVED** [review]

**Comment by leroyvn** [comment]
Looks good, we can proceed :)

**Pull request merged**